lib/middleware/favicon.js

```diff
@@ -1,47 +1,52 @@
 const fs = require('fs');
 const path = require('path');
-const Utils = require('../utils');
+
+const FAVICON_PATH = '/favicon.ico';
 
 /**
- * Returns the first favicon found in the supplied paths
+ * Returns the first favicon found in the supplied paths, if any.
+ *
+ * @param {array} [paths=[]]
+ * @return {string|false}
+ */
+function findFavicon(paths = []) {
+  for (const p of paths) {
+    const filePath = path.join(p, FAVICON_PATH);
+    if (fs.existsSync(filePath)) { return filePath; }
+  }
+  return false;
+}
+
+/**
+ * Serves the first favicon found in the supplied paths
  *
  * @param {array} [paths=[]]
  * @parms {Object} options
  * @return {function|boolean}
  */
 module.exports = function favicon(paths = [], options = {}) {
-  const faviconPath = '/favicon.ico';
   const maxAge = options.maxAge === null
     ? 86400000
     : Math.min(Math.max(0, options.maxAge), 31556926000);
   const cacheControl = `public, max-age=${maxAge / 1000 | 0}`; // eslint-disable-line no-bitwise
-  let icon;
-  let filePath;
 
   return (ctx, next) => {
-    if (ctx.path !== faviconPath) {
+    if (ctx.path !== FAVICON_PATH) {
       return next();
     }
 
     if (ctx.method !== 'GET' && ctx.method !== 'HEAD') {
       ctx.status = ctx.method === 'OPTIONS' ? 200 : 405;
       ctx.set('Allow', 'GET, HEAD, OPTIONS');
     } else {
-      Utils.each(paths, (p) => {
-        filePath = path.join(p, faviconPath);
-        try {
-          icon = fs.readFileSync(filePath);
-        } catch (err) { /* Do nothing */ }
-        if (icon) return false;
-
-        return true;
-      });
-
+      const filePath = findFavicon(paths);
       ctx.set('Cache-Control', cacheControl);
       ctx.type = 'image/x-icon';
-      ctx.body = icon;
+      ctx.body = filePath ? fs.readFileSync(filePath) : '';
     }
 
     return true;
   };
 };
+
+module.exports.findFavicon = findFavicon;
```

Comment:
This feels weird, but I couldn't come up with a better way to share this business logic between `VapidServer` and `VapidBuilder` without a larger utils refactor.